hermes-common/src/main/java/pl/allegro/tech/hermes/common/message/wrapper/AvroMessageContentWrapper.java

```diff
@@ -64,8 +65,11 @@ private MessageMetadata getMetadata(Map<Utf8, Utf8> metadata) {
             GenericRecord genericRecord = bytesToRecord(message, schema);
             genericRecord.put(METADATA_MARKER, metadataMap(id, timestamp, externalMetadata));
             return recordToBytes(genericRecord, schema);
-        } catch (Exception exception) {
-            throw new WrappingException("Could not wrap avro message", exception);
+        } catch (Exception e) {
+            if (e instanceof AvroRuntimeException && e.getMessage().equals("Not a valid schema field: __metadata")) {
+                throw new AvroInvalidMetadataException("Avro schema has invalid field __metadata", e);
```

Comment:
I would change from `Avro schema has invalid field __metadata` to `Schema does not contain mandatory __metadata field for Hermes internal metadata. Please fix topic schema.` - user will have precise information about the problem and how to fix it.  Other than that: +1